cpp/src/ice2slice/Main.cpp

```diff
@@ -167,7 +167,10 @@ compile(const vector<string>& argv)
         {
             PreprocessorPtr preprocessor = Preprocessor::create(argv[0], fileName, preprocessorArgs);
             FILE* preprocessedHandle = preprocessor->preprocess("-D__ICE2SLICE__");
-            assert(preprocessedHandle);
+            if (preprocessedHandle == nullptr)
```

Comment:
I think it would be nicer to fix `preprocess` to throw instead of returning null.